Mathlib/RingTheory/TwoSidedIdeal/Lattice.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import Mathlib.RingTheory.TwoSidedIdeal.Basic
+
+/-!
+# The complete lattice structure on two-sided ideals
+-/
+
+namespace TwoSidedIdeal
+
+variable (R : Type*) [NonUnitalNonAssocRing R]
+
+instance : Sup (TwoSidedIdeal R) where
+  sup := fun I J => { ringCon := I.ringCon ⊔ J.ringCon }
+
+instance : Inf (TwoSidedIdeal R) where
+  inf := fun I J => { ringCon := I.ringCon ⊓ J.ringCon }
+
+lemma mem_inf {I J : TwoSidedIdeal R} {x : R} :
+    x ∈ I ⊓ J ↔ x ∈ I ∧ x ∈ J :=
+  Iff.rfl
+
+instance : SupSet (TwoSidedIdeal R) where
+  sSup := fun s => { ringCon := sSup $ TwoSidedIdeal.ringCon '' s }
+
+instance : InfSet (TwoSidedIdeal R) where
+  sInf := fun s => { ringCon := sInf $ TwoSidedIdeal.ringCon '' s }
+
+lemma mem_iInf {ι : Type*} {I : ι → TwoSidedIdeal R} {x : R} :
+    x ∈ iInf I ↔ ∀ i, x ∈ I i :=
+  show (∀ _, _) ↔ _ by simp [mem_iff]
+
+lemma mem_sInf {S : Set (TwoSidedIdeal R)} {x : R} :
+    x ∈ sInf S ↔ ∀ I ∈ S, x ∈ I :=
+  show (∀ _, _) ↔ _ by simp [mem_iff]
+
+instance : Top (TwoSidedIdeal R) where
+  top := { ringCon := ⊤ }
+
+instance : Bot (TwoSidedIdeal R) where
+  bot := { ringCon := ⊥ }
+
+lemma mem_bot {x : R} : x ∈ (⊥ : TwoSidedIdeal R) ↔ x = 0 :=
+  Iff.rfl
+
+instance : CompleteLattice (TwoSidedIdeal R) :=
+  Function.Injective.completeLattice (f := TwoSidedIdeal.ringCon (R := R))
+    ringCon_injective (fun _ _ => rfl) (fun _ _ => rfl)
+      (fun _ => sSup_image) (fun _ => sInf_image) rfl rfl
+
```

Comment:
This actually causes instance diamond: we now have two partial orders on `TwoSidedIdeals`: one from `SetLike` instances; the other from `Lattice` structure. I need some help on solving this.

Reply:
One potential solution is to write this instance by hand.